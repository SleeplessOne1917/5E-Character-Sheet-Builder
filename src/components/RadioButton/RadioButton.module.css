.radio-button {
    width: 1.5rem;
    height: 1.5rem;
    border-radius: 50%;
    background-color: var(--heaven-light);
    border: 3px solid var(--davy);
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
}

.inner-circle {
    width: 0.8rem;
    height: 0.8rem;
    border-radius: 50%;
    background-color: var(--davy);
    transform: scale(0);
    transition: transform 0.2s ease-in;
}

.selected.radio-button {
    background-color: var(--highlight) !important;
}

.selected .inner-circle {
    transform: scale(1);
}

.radio-label {
	display: grid;
	font-size: 1.3rem;
	grid-template-columns: auto auto;
	gap: 0.5rem;
	cursor: pointer;
	user-select: none;
    margin: 0.5rem;
}

@media (hover: hover) {
    .radio-button:hover , .radio-label:hover .radio-button {
        background-color: var(--highlight-light);
    }
}