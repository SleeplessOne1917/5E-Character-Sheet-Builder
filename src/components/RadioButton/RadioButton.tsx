import { KeyboardEventHandler, useCallback } from 'react';

import classes from './RadioButton.module.css';
import { handleKeyDownEvent } from '../../services/handlerService';

type RadioButtonProps = {
	value: string;
	selected: string;
	labelText: string;
	onChange: (value: string) => void;
};

const RadioButton = ({
	value,
	selected,
	onChange,
	labelText
}: RadioButtonProps) => {
	const handleClick = useCallback(() => {
		onChange(value);
	}, [onChange, value]);

	const handleKeyDown: KeyboardEventHandler<HTMLDivElement> = useCallback(
		e => {
			handleKeyDownEvent(e, handleClick);
		},
		[handleClick]
	);

	return (
		<label
			className={classes['radio-label']}
			onClick={handleClick}
			data-testid="radio-button"
		>
			<div
				className={`${classes['radio-button']}${
					value === selected ? ` ${classes.selected}` : ''
				}`}
				tabIndex={0}
				role="radio"
				aria-checked={value === selected}
				onClick={handleClick}
				onKeyDown={handleKeyDown}
			>
				<div className={classes['inner-circle']} />
			</div>
			{labelText}
		</label>
	);
};

export default RadioButton;
