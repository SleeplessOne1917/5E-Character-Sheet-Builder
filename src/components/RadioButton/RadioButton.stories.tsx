import { ComponentMeta, ComponentStory } from '@storybook/react';

import RadioButton from './RadioButton';

export default {
	title: 'Components/RadioButton',
	component: RadioButton,
	args: {
		selected: 'bar',
		value: 'foo',
		labelText: 'Foo'
	}
} as ComponentMeta<typeof RadioButton>;

const Template: ComponentStory<typeof RadioButton> = args => (
	<RadioButton {...args} />
);

export const Default = Template.bind({});

export const Checked = Template.bind({});

Checked.args = {
	selected: 'foo'
};
