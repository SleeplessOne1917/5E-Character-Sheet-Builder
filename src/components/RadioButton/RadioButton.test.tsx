import * as stories from './RadioButton.stories';

import { render, screen } from '@testing-library/react';

import { composeStories } from '@storybook/testing-react';

const { Checked, Default } = composeStories(stories);

describe('renders correctly', () => {
	it('when unchecked', () => {
		render(<Default />);

		expect(screen.getByTestId('radio-button')).toMatchSnapshot();
	});

	it('when checked', () => {
		render(<Checked />);

		expect(screen.getByTestId('radio-button')).toMatchSnapshot();
	});
});
