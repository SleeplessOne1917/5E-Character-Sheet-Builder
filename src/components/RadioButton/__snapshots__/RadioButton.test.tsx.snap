// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when checked 1`] = `
<label
  class="radio-label"
  data-testid="radio-button"
>
  <div
    aria-checked="true"
    class="radio-button selected"
    role="radio"
    tabindex="0"
  >
    <div
      class="inner-circle"
    />
  </div>
  Foo
</label>
`;

exports[`renders correctly when unchecked 1`] = `
<label
  class="radio-label"
  data-testid="radio-button"
>
  <div
    aria-checked="false"
    class="radio-button"
    role="radio"
    tabindex="0"
  >
    <div
      class="inner-circle"
    />
  </div>
  Foo
</label>
`;
